CoreScriptsRoot/Modules/Server/ClientChat/BubbleChat/BubbleChat.lua

```diff
@@ -0,0 +1,693 @@
+
+	if sourcePlayer and line.Origin then
+		local fifo = this.CharacterSortedMsg:Get(line.Origin).Fifo
+		fifo:PushBack(line)
+		--Game chat (badges) won't show up here
+		this:CreateChatLineRender(sourcePlayer.Character, line, true, fifo)
+	end
+end
+
+function this:OnGameChatMessage(origin, message, color)
+	local localPlayer = PlayersService.LocalPlayer
+	local fromOthers = localPlayer ~= nil and (localPlayer.Character ~= origin)
+
+	local bubbleColor = BubbleColor.WHITE
+
+	if color == Enum.ChatColor.Blue then bubbleColor = BubbleColor.BLUE
+	elseif color == Enum.ChatColor.Green then bubbleColor = BubbleColor.GREEN
+	elseif color == Enum.ChatColor.Red then bubbleColor = BubbleColor.RED end
+
+	local safeMessage = this:SanitizeChatLine(message)
+	local line = createGameChatLine(origin, safeMessage, not fromOthers, bubbleColor)
+
+	this.CharacterSortedMsg:Get(line.Origin).Fifo:PushBack(line)
+	this:CreateChatLineRender(origin, line, false, this.CharacterSortedMsg:Get(line.Origin).Fifo)
+end
+
+function this:BubbleChatEnabled()
+	local clientChatModules = ChatService:FindFirstChild("ClientChatModules")
+	if clientChatModules then
+		local chatSettings = require(clientChatModules:WaitForChild("ChatSettings"))
+		if chatSettings.BubbleChatEnabled ~= nil then
+			return chatSettings.BubbleChatEnabled
+		end
+	end
+	return PlayersService.BubbleChat
+end
+
+function this:ShowOwnFilteredMessage()
+	local clientChatModules = ChatService:FindFirstChild("ClientChatModules")
+	if clientChatModules then
+		local chatSettings = require(clientChatModules:WaitForChild("ChatSettings"))
+		return chatSettings.ShowOwnFilteredMessage
+	end
+	return false
+end
+
+function this:CameraChanged(prop)
+	if prop == "CoordinateFrame" then
+		this:CameraCFrameChanged()
+	end
+end
+
+
+function findPlayer(playerName)
+	for i,v in pairs(PlayersService:GetPlayers()) do
+		if v.Name == playerName then
+			return v
+		end
+	end
+end
+
+ChatService.Chatted:connect(function(origin, message, color) this:OnGameChatMessage(origin, message, color) end)
+
+local cameraChangedCon = nil
+if game.Workspace.CurrentCamera then
+	cameraChangedCon = game.Workspace.CurrentCamera.Changed:connect(function(prop) this:CameraChanged(prop) end)
+end
+
+game.Workspace.Changed:connect(function(prop)
+	if prop == "CurrentCamera" then
+		if cameraChangedCon then cameraChangedCon:disconnect() end
+		if game.Workspace.CurrentCamera then
+			cameraChangedCon = game.Workspace.CurrentCamera.Changed:connect(function(prop) this:CameraChanged(prop) end)
+		end
+	end
+end)
+
+local ChatEvents = ReplicatedStorage:WaitForChild("DefaultChatSystemChatEvents")
+local OnMessageDoneFiltering = ChatEvents:WaitForChild("OnMessageDoneFiltering")
+local OnNewMessage = ChatEvents:WaitForChild("OnNewMessage")
+
+OnNewMessage.OnClientEvent:connect(function(messageData, channelName)
+	local sender = findPlayer(messageData.FromSpeaker)
+	if not sender then
+		return
+	end
+
+	if not messageData.IsFiltered then
```

Comment:
Nice spot. I have added a fix for this. I tested this but I actually wasn't reading ShowUserOwnFilteredMessage correctly, so it was always false.
https://github.com/ROBLOX/Core-Scripts/pull/673/commits/95fcb45dc5b5c4b4d424c602f503a9be3f1855f0